### PR TITLE
Added Structured Table Type

### PR DESCRIPTION
**Pull request by mybabysexy**
Implement #101

### CODE REVIEW

**Comment by mybabysexy** [comment]
I have read the CLA Document and I hereby sign the CLA
recheck

**Comment by mybabysexy** [comment]
recheck

**Comment by mybabysexy** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by mybabysexy** [comment]
recheck

File: src/Mapping/Operator/Factory/AsStructuredTable.php [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+    /**
+     * @param string $inputType
+     * @param int|null $index
+     *
+     * @return string
+     */
+    public function evaluateReturnType(string $inputType, int $index = null): string
+    {
+        return TransformationDataTypeService::DEFAULT_ARRAY;
```
Comment (fashxp) [inline src/Mapping/Operator/Factory/AsStructuredTable.php:64]: we would need to introduce there an extra type which just allows assignment to structured table fields, like we have it for quantity value, gallery etc.

File: src/Mapping/Type/TransformationDataTypeService.php [hunk 2]
```diff
@@ -67,7 +67,9 @@ class TransformationDataTypeService
         self::DEFAULT_ARRAY => [
             'multiselect',
             'countrymultiselect',
-            'languages'
+            'languages',
+            'table',
+            'structuredTable'
```
Comment (fashxp) [inline src/Mapping/Type/TransformationDataTypeService.php:72]: should be moved to an extra type...

File: src/Mapping/Operator/Factory/AsStructuredTable.php [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    public function process($inputData, bool $dryRun = false)
+    {
+        if (!is_array($inputData)) {
+            $inputData = [$inputData];
+        }
+        if ($dryRun) {
+            if (empty($inputData)) {
+                return "";
+            }
+            return (new StructuredTable($inputData))->getHtmlTable(
```
Comment (fashxp) [inline src/Mapping/Operator/Factory/AsStructuredTable.php:39]: Why return a html table here? 
Shouldn't be that in `generateResultPreview`, as for example [here](https://github.com/pimcore/data-importer/blob/43fb4128dac8284fcfc29b7ec6d2ffed99c4d22b/src/Mapping/Operator/Factory/Gallery.php#L73)

File: src/Mapping/Operator/Factory/AsStructuredTable.php [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+                array_map(function ($value, $label) {
+                    return [
+                        'key' => $value,
+                        'label' => $label,
+                    ];
+                }, array_keys($inputData), array_keys(array_values($inputData)[0])),
```
Comment (fashxp) [inline src/Mapping/Operator/Factory/AsStructuredTable.php:45]: `$inputData` might be empty/null ... will result in errors at this place.

**Review by fashxp — CHANGES_REQUESTED** [review]
Thx very much. We would need some tweaks here and there though.

**Comment by fashxp** [comment]
@mybabysexy :ping_pong: did you already have a chance to take a look at my comments?

**Comment by fashxp** [comment]
@mybabysexy 🏓

**Comment by fashxp** [comment]
@mybabysexy :ping_pong: any chance we could continue the discussion here?

**Comment by mybabysexy** [comment]
> @mybabysexy 🏓 any chance we could continue the discussion here?

hi, sorry I'm no longer working with Pimcore or PHP so I don't think that I can continue with this PR

**Comment by fashxp** [comment]
> > @mybabysexy 🏓 any chance we could continue the discussion here?
> 
> hi, sorry I'm no longer working with Pimcore or PHP so I don't think that I can continue with this PR

thx for the information.

**Comment by fashxp** [comment]
closing due to inactivity

**Pull request closed without merging**